wrappers/src/object_cs.cpp

```diff
@@ -241,11 +241,16 @@ extern "C" {
         });
     }
 
-    REALM_EXPORT int64_t object_get_obj_key(const Object& object, NativeException::Marshallable& ex)
+    REALM_EXPORT int32_t object_get_hashcode(const Object& object, NativeException::Marshallable& ex)
     {
         return handle_errors(ex, [&]() {
-            // ObjKey is incompatible with C, so we return just the value.
-            return object.obj().get_key().value;
+            auto table_key_value = object.obj().get_table()->get_key().value;
+            auto object_key_value = object.obj().get_key().value;
+
+            int32_t hashCode = -9999999769;
```

Comment:
That warning is correct - `-9999999769` is beyond what `int32` can hold. This is what VS generated for a simple struct with table key and object key if you want to use it:

```csharp
            public override int GetHashCode()
            {
                var hashCode = -986587137;
                hashCode = hashCode * -1521134295 + ObjKey.GetHashCode();
                hashCode = hashCode * -1521134295 + TableKey.GetHashCode();
                return hashCode;
            }
```